lib/providers/kms.js

```diff
@@ -17,12 +17,16 @@ class KMSProvider {
   constructor(secret) {
     checkNotEmpty(secret, 'secret is required');
     checkNotEmpty(secret.ciphertext, 'secret.ciphertext is required');
+    checkNotEmpty(secret.region, 'secret.region is required');
```

Comment:
Requiring the `secret.region` parameter breaks old Propsd clients that don't provide a region with their secret. To maintain backwards compatibility with old clients, we can do the following:

1. Use the `secret.region` parameter if it's provided.
2. Use the `Config.get('kms:region')` if there's no `secret.region` parameter.
3. Use `us-east-1` if there's no config parameter.

We should check that provided regions aren't `null`, `undefined`, or an empty string.